src/specs/networking.md

```diff
@@ -66,3 +71,40 @@ Accepting a `WireTxPayForMessage` into the mempool requires different logic than
 
 Receiving a `WireTxPayForMessage` object from the network follows the reverse process: for each `message_commitment_and_signature`, verify using the [based on the non-interactive default rules](../rationale/message_block_layout.md#non-interactive-default-rules) that the signature is valid.
+
+## Invalid Erasure Coding
+
+If a malicious block producer incorrectly computes the 2D Reed-Solomon code for a block's data, a fraud proof for this can be presented. We assume that the light clients have the [AvailableDataHeader](./data_structures.md#availabledataheader) and the [Header](./data_structures.md#header) for each block. Hence, given a [ShareProof](#shareproof), they can verify if the `rowRoot` or `colRoot` specified by `isCol` and `position` commits to the corresponding [Share](./data_structures.md#share). Similarly, given the `height` of a block, they can access all elements within the [AvailableDataHeader](./data_structures.md#availabledataheader) and the [Header](./data_structures.md#header) of the block. 
+
+### ShareProof
+
+| name       | type                                                                                        | description                                                                                      |
+|------------|---------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------|
+| `share`    | [Share](./data_structures.md#share)                                                         | The share.                                                                                        |
+| `proof`    | [NamespaceMerkleTreeInclusionProof](./data_structures.md#namespacemerkletreeinclusionproof) | The Merkle proof of the share in the offending row or column root.                                |
+| `isCol`    | `bool`                                                                                      | A Boolean indicating if the proof is from a row root or column root; `false` if it is a row root. |
+| `position` | `uint64`                                                                                    | The index of the share in the offending row or column.                                            |
+
+
+### BadEncodingFraudProof
+
+| name          | type                                                                                  | description                                                                                           |
```

Comment:
For these, can you additionally create protobuf definitions, since they're sent over the network? You can include code inline from other files (see example later in this file).

Reply:
Created the protobuf definitions, added them inline.